server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -325,6 +438,83 @@ public RecoverySettings(Settings settings, ClusterSettings clusterSettings) {
         );
     }
 
+    private void computeMaxBytesPerSec(Settings settings) {
+        validateBandwidthsSettings(settings);
```

Comment:
We'll run this as a cluster state applier so we can't throw exceptions if we're in an invalid state somehow. We can definitely assert that they're valid, but in production we must apply whatever we have.

Reply:
Good catch 🤦🏻 

I renamed to `validateNodeBandwidthRecoverySettings` and moved the method in the constructor in 8d71b8bb62ae9d7bdee8c1de45934bc86270df2f.